<launch>

  <arg name="use_rosbag" default="false" />
  <arg name="filename" default="/" />
  <arg name="gui" default="true"/>
  <arg name="target_class" default="" />
  <arg name="save_when_action" default="true"/>
  <arg name="threshold" default="0.5"/>
  <arg name="save_data_rate" default="5"/>

  <include file="$(find decopin_hand)/launch/audio_to_spectrogram.launch" >
    <arg name="use_rosbag" value="$(arg use_rosbag)" />
    <arg name="filename" value="$(arg filename)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="threshold" value="$(arg threshold)" />
  </include>

  <!-- Collect spectrogram with action class, only when the robot is in action. -->
  <node pkg="decopin_hand" type="action_saver.py" name="action_saver" output="screen">
    <remap from="~in_action" to="/action_detector_volume/in_action" />
    <remap from="~input" to="/preprocess_gray_image/output_normalized" />
    <remap from="~input_raw" to="/spectrum_to_spectrogram/spectrogram" />
    <rosparam subst_value="true">
      <!-- params below is enabled only when save_data is true -->
      save_data_rate: $(arg save_data_rate) <!-- related to spectrogram_period -->
      target_class: $(arg target_class)
      save_when_action: $(arg save_when_action)
      save_raw_spectrogram: true
    </rosparam>
  </node>

</launch>
